library/kernel_settings.py

```diff
@@ -0,0 +1,777 @@
+    the params to profile"""
+    errlist = []
+    replaces = {}
+    for sectionname, items in params.items():
+        if isinstance(items, dict):
+            if not items == REMOVE_SECTION_VALUE:
+                errlist.append(
+                    ERR_REMOVE_SECTION_VALUE.format(sectionname, REMOVE_SECTION_VALUE)
+                )
+        elif isinstance(items, list):
+            for item in items:
+                if not item:
+                    continue # ignore empty items
+                if not isinstance(item, dict):
+                    errlist.append(ERR_ITEM_NOT_DICT.format(sectionname, item))
+                elif "previous" in item:
+                    if item["previous"] != "replaced":
+                        errlist.append(
+                            ERR_ITEM_PREVIOUS.format(sectionname, item["previous"])
+                        )
+                    else:
+                        replaces[sectionname] = True
+                else:
+                    itemerrlist = validate_and_digest_item(sectionname, item)
+                    errlist.extend(itemerrlist)
+        else:
+            errlist.append(ERR_ITEM_DICT_OR_LIST.format(sectionname))
+    if replaces:
+        params[SECTION_TO_REPLACE] = replaces
+
+    return errlist
+
+
+def run_module():
+    """ The entry point of the module. """
+
+    module_args = dict(
+        name=dict(type="str", required=False),
+        purge=dict(type="bool", required=False, default=False),
+    )
+    tuned_plugin_names = get_supported_tuned_plugin_names()
+    for plugin_name in tuned_plugin_names:
+        # use raw here - type can be dict or list - perform validation
+        # below
+        module_args[plugin_name] = dict(type="raw", required=False)
+
+    result = dict(changed=False, original_message="", message="")
+
+    module = AnsibleModule(argument_spec=module_args, supports_check_mode=True)
+
+    if os.environ.get("TESTING", "false") == "true":
+        _ = setup_for_testing()
+
+    params = module.params
+    # remove any non-tuned fields from params and save them locally
+    # state = params.pop("state")
+    purge = params.pop("purge", False)
+    paramname = params.pop("name")
+    # also remove any empty or None
+    for key, val in list(params.items()):
+        if not val:
+            del params[key]
+    errlist = validate_and_digest(params)
+    if errlist:
+        errmsg = "Invalid format for input parameters"
+        module.fail_json(msg=errmsg, warnings=errlist, **result)
+
+    tuned_config = get_tuned_config()
+    current_profile = None
+    tuned_app, errmsg = load_current_profile(tuned_config, TUNED_PROFILE, module)
+
+    if tuned_app is None:
+        module.fail_json(msg=errmsg, **result)
+    else:
+        current_profile = tuned_app.daemon.profile
+        debug_print_profile(current_profile, module)
+        errmsg = ""
+
+    # if the user is working with this module in only check mode we do not
```

Comment:
It's unclear to me whether the check mode does anything useful given the changes in 338afd5695b.

Reply:
good catch - fixed